service/db.go

```diff
@@ -462,58 +462,67 @@ func updateUserRow(uuid string, svcDerived *pblib.User, dbDerived *pblib.User) (
 	return updatedUser, nil
 }
 
-//TODO work on later
-func insertNewSecret(secret *pblib.Secret) error {
-	if secret == nil {
-		return authconst.ErrNilSecret
+// insertNewSecret inserts a newly generated secret key to database
+// secret key is used to sign JWT's
+// Returns err if secret is empty or error with database
+func insertNewSecret() error {
+	// generate a new secret
+	secretKey, err := generateToken(auth.SecretByteSize)
```

Comment:
Based on your variable name, it is better to rename `generateToken` as `generateSecretKey`

Reply:
i am also using `generateToken` for generating email token too, still change the name?

Reply:
yes